pkg/controller/ready.go

```diff
@@ -20,17 +21,14 @@ func (c *Controller) ensureResourceReady(ctx context.Context, res *Resource) err
 	span, ctx := opentracing.StartSpanFromContext(ctx, "checkReady")
 	defer span.Finish()
 
-	objectKey, err := client.ObjectKeyFromObject(res.resource)
-	if err != nil {
-		return serrors.UnrecoverrableError(err, serrors.OperatorReason, "cannot get object key")
-	}
+	objectKey := client.ObjectKeyFromObject(res.resource)
 
 	gvk := c.AddGVKToSpan(ctx, span, res.resource)
 	l := logger.Get(ctx)
 
-	result := res.resource.DeepCopyObject()
+	result := res.resource.DeepCopyObject().(resources.Resource)
```

Comment:
Why do we need to cast it?
Does the following not work?
```suggestion
	var result resources.resource = res.resource.DeepCopyObject()
```

Reply:
because DeepCopyObject() is returning running.Object not client.Object

Reply:
your suggestion code is not working